tests/connect_two.rs

```diff
@@ -0,0 +1,71 @@
+use async_std::task;
+
+/// Make sure two instances of ipfs can be connected.
+#[test]
+fn connect_two_nodes() {
+    // env_logger::init();
+
+    // make sure the connection will only happen through explicit connect
+    let mdns = false;
+
+    let (tx, rx) = futures::channel::oneshot::channel();
+
+    let node_a = task::spawn(async move {
+        let opts = ipfs::IpfsOptions::inmemory_with_generated_keys(mdns);
+        let (ipfs, fut) = ipfs::UninitializedIpfs::new(opts)
+            .await
+            .start()
+            .await
+            .unwrap();
+
+        let jh = task::spawn(fut);
+
+        let (pk, addrs) = ipfs
+            .identity()
+            .await
+            .expect("failed to read identity() on node_a");
+        assert!(!addrs.is_empty());
+        tx.send((pk, addrs, ipfs, jh)).unwrap();
+    });
+
+    task::block_on(async move {
+        let (other_pk, other_addrs, other_ipfs, other_jh) = rx.await.unwrap();
+
+        println!("got back from the other node: {:?}", other_addrs);
+
+        let opts = ipfs::IpfsOptions::inmemory_with_generated_keys(mdns);
+        let (ipfs, fut) = ipfs::UninitializedIpfs::new(opts)
+            .await
+            .start()
+            .await
+            .unwrap();
+        let jh = task::spawn(fut);
+
+        let _other_peerid = other_pk.into_peer_id();
+
+        let mut connected = None;
+
+        for addr in other_addrs {
+            println!("trying {}", addr);
+            match ipfs.connect(addr.clone()).await {
+                Ok(_) => {
+                    connected = Some(addr);
+                    break;
+                }
+                Err(e) => {
+                    println!("Failed connecting to {}: {}", addr, e);
+                }
+            }
```

Comment:
My initial thinking for the `connect` was that it could be `connect(PeerId, Vec<Multiaddr>)` as by the looks of libp2p it might dial through all of the addresses. This will probably change somehow in the next version of libp2p because the multiple connections to a peer was merged https://github.com/libp2p/rust-libp2p/pull/1440 and https://github.com/libp2p/rust-libp2p/pull/1493.